ext/posix/tests/posix_setegid_basic.phpt

```diff
@@ -0,0 +1,12 @@
+--TEST--
+Test function posix_setegid() by calling it with its expected arguments
+--SKIPIF--
+<?php 
+    if(!extension_loaded("posix")) die("skip - POSIX extension not loaded"); 
+?>
+--FILE--
+<?php
+var_dump(posix_setegid(posix_getegid()));
```

Comment:
@andrewnester it needs [root privileges](http://php.net/manual/en/function.posix-setegid.php#refsect1-function.posix-setegid-description) to actually change the group ID.

```php
// content of simple posix_test.php
echo 'My real group id is '.posix_getgid() . PHP_EOL; //20
posix_setegid(40);
echo 'My real group id is '.posix_getgid() . PHP_EOL; //20
echo 'My effective group id is '.posix_getegid() . PHP_EOL; //40
```
run it without sudo
```bash
➜  php posix_test.php
My real group id is 20
My real group id is 20
My effective group id is 20
```

run it with sudo
```bash
➜  sudo php posix_test.php
My real group id is 0
My real group id is 0
My effective group id is 40
```
How can I tell the test runner to run it with root privilege, any specific section I need to mention in the test?

I see most of the [posix_set*_basic](https://github.com/php/php-src/blob/master/ext/posix/tests/posix_setgid_basic.phpt#L17-L18) tests are usually doing the same pattern, thoughts?